db/database.go

```diff
@@ -303,9 +302,26 @@ func NewDatabaseContext(dbName string, bucket base.Bucket, autoImport bool, opti
 
 			// TODO: invoke the same callback function from there as well, to pick up the auto-online handling
 
-		}); err != nil {
+		})
+
+		if err != nil {
+			// In the case of an error, unlock the change cache despite the fact it's in an unitialized state.
+			context.changeCache.StartupUnlock()
```

Comment:
Looking at this again, I'm still not thrilled about trying to manage the changeCache's internal lock from an external location.  Would this approach work:
 - changeCache.Init() leaves the cache locked (not started)
 - A subsequent call to changeCache.Start() does the following:
    - retrieves and sets initial/next sequence
    - unlocks the cache
This would also let us push everything from line 313 to 324 down into changeCache.Start(), which I think makes more sense (it's cache-related code).

Reply:
Yeah, agreed that it's awkward to have the external locking semantics.  Fixed in 8ee61c51fac45632f3b3132c928b383288beb2be